lottie/src/main/java/com/airbnb/lottie/L.java

```diff
@@ -75,10 +75,16 @@ private static LottieTrace getTrace() {
 
   public static void setFetcher(LottieNetworkFetcher customFetcher) {
     fetcher = customFetcher;
+    synchronized (NetworkCache.class) {
+      networkFetcher = null;
```

Comment:
What is the purpose of synchronizing a single line?

Reply:
In both cases here, it's probably best to only reset the value if the new value is different so that calling this is idempotent.

Reply:
I was trying to guard against a race between this and the `networkFetcher(Context)` function:
```java
NetworkFetcher local = networkFetcher;
if (local == null) {
  synchronized (NetworkFetcher.class) {
    local = networkFetcher;
    if (local == null) {
      networkFetcher = local = new NetworkFetcher(networkCache(context), fetcher != null ? fetcher : new DefaultLottieNetworkFetcher());
    }
    // What if networkFetcher gets set to null right now?
  }
}
return local;
```

In writing it out explicitly I realize it doesn't matter: in this case the next invocation of `networkFetcher(Context)` will generate a new instance as expected. I'll drop the new `synchronized` calls.